docs: add operator documentation (#3406)
inspektor-gadget/inspektor-gadget: docs/spec/operators/ebpf.md

---
title: ebpf
sidebar_position: 10
---

The eBPF operator handles the [ebpf layer](../oci.md#the-ebpf-layer)
that contains an ELF file with the eBPF programs. This operator loads the
programs into the kernel and attaches them to the different hooks as specified
by the gadget developer.

## Instance Parameters

### `iface`

> Network interface to attach networking programs to.

Reviewer: ```suggestion
Network interface to attach networking programs to. Only available if the gadget uses networking programs.
```

Not sure if it's worth mentioning that.

Fully qualified name: `operators.oci.ebpf.iface`

### `trace-pipe`

Print debug information generated by eBPF with `bpf_printk()` to the terminal.

Fully qualified name: `operators.oci.ebpf.trace-pipe`

### `map-fetch-interval`

> Interval in which to iterate over "iter maps", i.e. hashmaps marked with
> `GADGET_MAPITER()`

Reviewer: ```suggestion
Interval in which to iterate over eBPF maps that have been marked with
`GADGET_MAPITER()`
```

Fully qualified name: `operators.oci.ebpf.map-fetch-interval`

Default: `1000ms`

### `map-fetch-count`

> Number fetch cycles (use 0 for unlimited) for "iter maps", i.e. hashmaps marked
> with `GADGET_MAPITER()`.

Reviewer: ```suggestion
Number of fetch cycles (use 0 for unlimited) for eBPF maps that have been marked with `GADGET_MAPITER()`.
```
